firefox-ios/firefox-ios-tests/Tests/XCUITests/AddressesTests.swift

```diff
@@ -0,0 +1,211 @@
+        // The address is saved
+        mozWaitForElementToExist(app.staticTexts[savedAddressesTxt])
+        mozWaitForElementToExist(app.staticTexts["AL"])
+    }
+
+    // https://mozilla.testrail.io/index.php?/cases/view/2618645
+    func testAddNewAddressEmailFilled()  throws {
+        if #unavailable(iOS 16) {
+            throw XCTSkip("Addresses setting is not available for iOS 15")
+        }
+        reachAddNewAddressScreen()
+        // Enter a valid date for Full Name and press save
+        typeEmail(email: "test@mozilla.com")
+        tapSave()
+        // The "Address saved" toast message is displayed
+        mozWaitForElementToExist(app.staticTexts[addressSavedTxt])
+        // The address is saved
+        mozWaitForElementToExist(app.staticTexts[savedAddressesTxt])
+        mozWaitForElementToExist(app.staticTexts["AL"])
+    }
+
+    private func reachAddNewAddressScreen() {
+        navigator.nowAt(NewTabScreen)
+        waitForTabsButton()
+        navigator.goto(AddressesSettings)
+        let addresses = AccessibilityIdentifiers.Settings.Address.Addresses.self
+        mozWaitForElementToExist(app.navigationBars[addresses.title])
+        app.buttons[addresses.addAddress].tap()
+        mozWaitForElementToExist(app.navigationBars[addresses.addAddress])
+        if iPad() {
+            app.buttons["Close"].tap()
+            app.buttons[addresses.addAddress].tap()
+        }
+        mozWaitForElementToExist(app.staticTexts["Name"])
+    }
+
+    private func addNewAddress() {
+        typeName(name: "Test")
+        typeOrganization(organization: "organization test")
+        typeStreetAddress(street: "test address")
+        typeCity(city: "city test")
+        typeZIP(zip: "123456")
+        typePhone(phone: "01234567")
+        typeEmail(email: "test@mozilla.com")
+    }
+
+    private func typeName(name: String) {
+        app.staticTexts["Name"].tap()
+        app.typeText(name)
+    }
+
+    private func typeOrganization(organization: String) {
+        app.staticTexts["Organization"].tap()
+        app.typeText(organization)
+    }
+
+    private func typeStreetAddress(street: String) {
+        app.staticTexts["Street Address"].tap()
+        app.typeText(street)
+    }
+
+    private func typeCity(city: String) {
+        app.staticTexts["City"].tap()
+        app.typeText(city)
+    }
+
+    private func typeZIP(zip: String) {
+        scrollToElement(app.staticTexts["ZIP Code"])
+        app.staticTexts["ZIP Code"].tap()
+        app.typeText(zip)
+    }
+
+    private func typePhone(phone: String) {
+        if app.buttons["Done"].isHittable {
+            app.buttons["Done"].tap()
+        }
+        app.staticTexts["Phone"].tapOnApp()
+        app.typeText(phone)
+    }
+
+    private func typeEmail(email: String) {
+        scrollToElement(app.staticTexts["Email"])
+        app.staticTexts["Email"].tap()
+        app.typeText(email)
+    }
+
+    private func tapSave() {
```

Comment:
Can we use `waitAndTap()`. We have now a methos that wait for an element and then tap on it.

Reply:
updated with waitAndTap()

Reply:
LGTM